test/run/test_stages.py

```diff
@@ -521,3 +521,23 @@ def test_ovf(self):
             assert capacity == str(json.loads(res.stdout)["virtual-size"])
             pop_size = ovf_tree_disk.attrib["{http://schemas.dmtf.org/ovf/envelope/1}populatedSize"]
             assert pop_size == str(os.stat(vmdk).st_size)
+
+
+    def test_dnf4_mark(self):
+        datadir = self.locate_test_data()
+        testdir = os.path.join(datadir, "stages", "dnf4.mark")
+
+        with self.osbuild as osb, tempfile.TemporaryDirectory(dir="/var/tmp") as outdir:
+            osb.compile_file(os.path.join(testdir, "tree.json"), exports=["tree"], output_dir=outdir)
+
+            tree = os.path.join(outdir, "tree")
+            assert os.path.isdir(tree)
+
+            # we're going to verify that packages in the tree are marked according to
+            r = subprocess.run(["dnf", "--installroot", tree, "repoquery", "--installed" "--qf", "%{name},%{reason}\n"],
```

Comment:
Missing comma between `installed` and `qf` args.

Reply:
Woa, how does DNF parse that correctly.

Reply:
Also, I think the `\n` at the end isn't needed (looking at my own output while running to test).

Reply:
> Woa, how does DNF parse that correctly.

I don't know.  It broke when I ran it myself.